From 930a336d3e5370ea71d75308f7928860dcc7020b Mon Sep 17 00:00:00 2001
From: Steve Kondik <shade@chemlab.org>
Date: Mon, 23 Dec 2013 13:33:21 -0800
Subject: [PATCH 021/206] camera: Check if video sizes are available

 * Some drivers don't report video sizes and we should avoid using fine
   resolution lookup, otherwise a crash results.

Change-Id: I8010e66d1a3a5e6f8a505ff3b50146cd24fc2b9c
---
 src/com/android/camera/CameraSettings.java | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/com/android/camera/CameraSettings.java b/src/com/android/camera/CameraSettings.java
index 9748dc2f0..7256be238 100755
--- a/src/com/android/camera/CameraSettings.java
+++ b/src/com/android/camera/CameraSettings.java
@@ -1364,8 +1364,9 @@ public class CameraSettings {
     private static ArrayList<String> getSupportedVideoQuality(int cameraId,Parameters parameters) {
         ArrayList<String> supported = new ArrayList<String>();
         // Check for supported quality
-        if (ApiHelper.HAS_FINE_RESOLUTION_QUALITY_LEVELS) {
-        getFineResolutionQuality(supported,cameraId,parameters);
+        if (ApiHelper.HAS_FINE_RESOLUTION_QUALITY_LEVELS &&
+                parameters.getSupportedVideoSizes() != null) {
+            getFineResolutionQuality(supported,cameraId,parameters);
         } else {
             supported.add(Integer.toString(CamcorderProfile.QUALITY_HIGH));
             CamcorderProfile high = CamcorderProfile.get(
-- 
2.17.1

